### DISCUSSION TITLE
Button Filter and Column not work for Bootstrap 5

### DISCUSSION THREAD

**Discussion by jarwonozt** [rappasoft/laravel-livewire-tables#1102] [Q&A]
I have a problem with the toolbar button Filter and Column button when using bootstrap 5, when I click there is no response and the dropdown does not appear. But for Bulk Action it works fine. Is it necessary to modify `toolbar.blade.php` ?

Livewire table class : https://pastebin.com/1Q0xMrZM

https://user-images.githubusercontent.com/67833034/225581285-b819554b-c481-40e5-8a33-bd583d9bdc38.mp4

**Comment by lrljoe** [comment]
Filters is definitely working in my demo instance for BS5.

Looking at your pastebin, your filter isn't actually setup to apply/do anything, and seems to have a list of the same items you're going to see in the main table.
Of course, you can do this, but I'd typically expect to see filtering based on something else (in your instance for example - the Author).

However I'll take your list of the same model as your desired outcome, see below

```php
    public function filters(): array
    {
        return [
            MultiSelectFilter::make('Post')
                ->options(
                    Post::query()
                        ->orderBy('title')
                        ->get()
                        ->keyBy('id')
                        ->map(fn($post) => $post->title)
                        ->toArray()
                ),
        ];
    }
```

Something like the below will actually do what it seems you're trying to do:
```php
    public function filters(): array
    {
        return [
            MultiSelectFilter::make('Post')
              ->options(
                  Post::query()
                      ->orderBy('title')
                      ->get()
                      ->keyBy('id')
                      ->map(fn($post) => $post->title)
                      ->toArray()
              )
              ->filter(function (Builder $builder, array $values) {
                        $builder->whereIn('id', $values);
              }),
        ];
    }
```

If this isn't working, can you check the DevTools console in your browser for any errors, both on loading the page, and on clicking the filters button.
Can you also confirm your Bootstrap version, and whether you've got all of the modules loaded, or just some?

**Reply by jarwonozt** [reply]
Thank you for answering, after I checked again and it turns out I haven't installed Aline.js
this is very important because it is mandatory to include Alpine.js, if anyone is experiencing the same thing, you can try adding `<script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn .min. js"></script>`

**Reply by lrljoe** [reply]
That'll do it!  


If you're all sorted then please mark this as closed, if you still have issues then reply and I'll try to lend a hand.
